### ISSUE TITLE
 no-shadowed-variable false positive

### ISSUE THREAD

**Issue by Fraegle** [palantir/tslint#650]
When defining a polymorphic interface such as
```JavaScript
interface EdgeCall<T, TResult> {
    (parameters: T, runSynchonous: boolean): TResult;
    (parameters: T, callback: (error: Error, result: TResult) => void): void;
}
```
(this interface wasn't my idea)
The second occurrence of parameters is wrongly detected by the rule.

**Comment by JKillian** [comment]
Thanks for the report, this should be fixed in a release within the next week or so
